### ISSUE TITLE
Low acc on PC Coder (GPS model) with E1 dataset

### ISSUE THREAD

**Issue by huangdi95** [shrivastavadisha/N-PEPS#1]
Hi! Thanks for your great work!

I have tried to reproduce the result of PC Coder (GPS model) on E1 but failed. The result in the Table 1 of paper is 77% in 5s. However, I only got 71.49% averaged across 30 test splits. The max of them is around 80%, and the min of them is only around 64%.

Can you provide me with more specific scripts such as `params.py` on E1 GPS and `test command for gps` to help me reproduce the result? Thanks!

I used the dataset downloaded from `README`:
```
348K    smallval_dataset_gps
8.7M    test_splits
62M     train_dataset_gps
89M     train_dataset_peps
6.9M    val_dataset_gps
9.9M    val_dataset_peps
```

`params.py`:
```
seed = 9
PE=False

# General training
num_epochs = 100
patience = 20
#model_output_path = 'trained_models/E1/PE_model' #'trained_models/E1/GPS_model/'
if PE:
    model_output_path = 'trained_models/E1/PE-'
else:
    model_output_path = 'trained_models/E1/GPS-'

load_from_checkpoint = None
max_len = None #For debugging set a lower number, use None for running training for full data

# File params (Inference)
global_model_path = 'trained_models/E1/GPS_model/best_model.th' #GPS model path
PE_model_path = 'trained_models/E1/PE_model/best_model.th' #PE model path
result_path = 'results/'

# GPS and PE model training params
if PE:
    train_path = 'data/E1/train_dataset_peps'
    val_path = 'data/E1/val_dataset_peps'
else:
    train_path = 'data/E1/train_dataset_gps'
    val_path = 'data/E1/val_dataset_gps'
learn_rate = 0.001
batch_size = 32
#batch_size = 8 
val_iterator_size = 32
lr_scheduler_step_size = 4

#Aggregator training params
att_batch_size = 100 #256

#Inference params
search_method = 'beam'
num_workers = 1
max_beam_size = 819200
dfs_max_width = 50
cab_beam_size = 100
cab_width = 10
cab_width_growth = 10

# DSL params
integer_min = -256
integer_max = 255
integer_range = integer_max - integer_min + 1
max_list_len = 20
num_inputs = 3
num_statements = 1298
num_operators = 38

# Program State Params
max_program_len = 8
max_program_vars = max_program_len + num_inputs
state_len = max_program_vars + 1
state_dim = 256

# H_theta and W_phi network params
type_vector_len = 2
embedding_size = 20
var_encoder_size = 56
dense_output_size = 256
dense_num_layers = 10
dense_growth_size = 56
```

`test command`:
```
python -u -m \
    scripts.solve_problems \
    --method "gps" \
    --agg_type "mean" \
    --agg_mode "none" \
    --key_type "sij" \
    --alpha 0.8 \
    --gps_timeout 5.0 \
    --peps_timeout 0.8 \
    --dropout 0.0 \
    --test_path "data/data/E1/test_splits/len_4/split_1" \
    --global_model_path ".//trained_models/E1/${GPS}/model-best" \
    --PE_model_path "./trained_models/E1/${PE}/model-best" \
    --result_path "results/E1/" \
    --att_model_path "" \
    --num_of_problems -1 \
    --max_program_len 4 
```

**Comment by shrivastavadisha** [comment]
Hey! Did you try with the trained GPS model for E1 provided with the repo? If not, did you use the same set of hyperparameters mentioned in Appendix C of the paper? Please check out Appendix C of our paper for complete details of the hyperparameters used. Also, note that to account for the variations in machines we ran the inference experiments across different machines and multiple runs on the same machine ( see Appendix C.5). All the inference results are carried out on CPUs only on a single thread.

**Comment by shrivastavadisha** [comment]
In addition, I am providing the individual test-split wise results for GPS for E1 that we got:

split-1 | split-2 | split-3 | split-4 | split-5 | split-6 | split-7 | split-8 | split-9 | split-10 | split-11 | split-12 | split-13 | split-14 | split-15 | split-16 | split-17 | split-18 | split-19 | split-20 | split-21 | split-22 | split-23 | split-24 | split-25 | split-26 | split-27 | split-28 | split-29 | split-30

78.4 | 78.6 | 81.4 | 79.2 | 73.2 | 80.2 | 76.8 | 78 | 76.8 | 74.2 | 75.8 | 77.8 | 76.6 | 79.2 | 76.6 | 76 | 81.8 | 78.4 | 76.6 | 78.8 | 78.6 | 76.8 | 75.8 | 76.4 | 78.2 | 74.2 | 77.8 | 80.4 | 79.8 | 80

**Comment by huangdi95** [comment]
I followed the training hyperparameters in Appendix C.1: `learning rate = 0.001; batch size = 32 for E1; lr_decay_epoch/rate = 4/0.1; Adam optimizer`.

I'll try the model in the repo to see if the problem happens while testing and report the result later, thanks!

**Comment by huangdi95** [comment]
Hi, I test the model in the repo and get 76.36 acc on average, is that reasonable?
Details:
76.8 | 76.8 | 80 | 77.8 | 72.4 | 77.8 | 75.8 | 77 | 75.6 | 73.6 | 75.2 | 76 | 75.4 | 78.4 | 74.8 | 74.6 | 80 | 77.6 | 75.8 | 76.4 | 77.8 | 74.4 | 74.6 | 74.6 | 77 | 72.2 | 76.8 | 78.6 | 79 | 78

I trained a new model based on the dataset provided in the repo, and got 75.11%.
Details:
75.4 | 74.8 | 76.4 | 78.8 | 72.4 | 75.6 | 74.8 | 76.2 | 75.2 | 75.6 | 76.2 | 72.4 | 70.2 | 75.8 | 75.6 | 71.4 | 80.2 | 76.8 | 76.8 | 75.4 | 76 | 73 | 76 | 73 | 74 | 73.4 | 72 | 76 | 76.2 | 77.6

**Comment by shrivastavadisha** [comment]
In my opinion, the numbers are reasonable accounting for the variation in machines and other factors. If your use case is just using PCCoder, you can directly use the codebase from https://github.com/amitz25/PCCoder. Our code is adapted from here only.
